### PR TITLE
Upload ODK submission user photos to S3 for easy access

### PR DESCRIPTION
**Pull request by Sujanadh**
## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue

- Issues : 
            - #1701 
            - #1729

## Describe this PR

This PR creates a function to upload submission images to s3
- run background task to upload images whenever submission table is opened
- checks if the object already exists in s3 and avoids task to run in background
- after successfully uploading images , it will create database record in `submission_photos` table with fields `project_id`, `task_id`, `submission_id`, and `s3_path`

## Screenshots

N/A

## Alternative Approaches Considered

Did you attempt any other approaches that are not documented in code?

## Review Guide

Notes for the reviewer. How to test this change?

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

### CODE REVIEW

File: src/backend/migrations/006-create-submission-photos-table.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+-- ## Migration to:
+-- * Create submission_photos table
+
+-- Start a transaction
+BEGIN;
+
+CREATE TABLE submission_photos (
```
Comment (spwoodcock) [inline src/backend/migrations/006-create-submission-photos-table.sql:7]: Check this SQL doesn't error if you run it twice 👍

Maybe `create table if not exists` will fix

**Review by spwoodcock — CHANGES_REQUESTED** [review]

File: src/backend/app/submissions/submission_crud.py [hunk 2]
```diff
@@ -565,3 +566,100 @@ async def get_submission_detail(
...
+            submission_detail = await get_submission_detail(instance_id, project, db)
+            attachments = submission_detail["verification"]["image"]
+
+            if not isinstance(attachments, list):
+                attachments = [attachments]
+
+            for idx, filename in enumerate(attachments):
+                s3_key = f"{s3_base_path}/{instance_id}/{idx+1}.jpeg"
+
+                if object_exists(s3_bucket, s3_key):
```
[the remark this answers is not in the document]
Comment (Sujanadh) [reply src/backend/app/submissions/submission_crud.py:606]: ah makes sense! :thinking:

File: src/backend/app/submissions/submission_crud.py [hunk 3]
```diff
@@ -565,3 +566,100 @@ async def get_submission_detail(
...
+                        add_obj_to_bucket(
+                            s3_bucket, image_stream, s3_key, content_type="image/jpeg"
+                        )
+
+                        # Generate the image URL
+                        img_url = f"{settings.S3_DOWNLOAD_ROOT}/{s3_bucket}/{s3_key}"
+
+                        # Insert the record into submission_photos table
+                        sql = text("""
+                        INSERT INTO submission_photos (
```
[the remark this answers is not in the document]
Comment (Sujanadh) [reply src/backend/app/submissions/submission_crud.py:632]: okay

File: src/backend/migrations/006-create-submission-photos-table.sql (hunk 1 above)
Comment (Sujanadh) [reply src/backend/migrations/006-create-submission-photos-table.sql:7]: okay

**Review by spwoodcock — APPROVED** [review]

**Comment by spwoodcock** [comment]
Looks good! Merging 💫

**Pull request merged**